### PR TITLE
Add RA service

### PR DESCRIPTION
**Pull request by Gwemox**
Registration Authority Service functionality.

### CODE REVIEW

File: Model/MatchingResult.php [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+     * @var string
+     */
+    protected $certificateStatus;
+
+    /**
+     * @var CertificateInfo
+     */
+    protected $certificateInfo;
+
+
```
Comment (konandrum) [inline Model/MatchingResult.php]: To remove (line break)

File: Model/MatchingResult.php [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+            ->setFirstname($resolvedOptions['subject'])
+            ->setLastname($resolvedOptions['issuer'])
+            ->setEmail($resolvedOptions['issuer'])
+            ->setCertificateLevel($resolvedOptions['issuer'])
+            ->setCertificateStatus($resolvedOptions['serial'])
+            ->setCertificateInfo($resolvedOptions['issuer'])
+        ;
+    }
+
+
```
Comment (konandrum) [inline Model/MatchingResult.php]: To remove (line break)

File: Model/ValidatorResult.php [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+     * @param OptionsResolver $resolver
+     */
+    public static function configureData(OptionsResolver $resolver)
+    {
+        $resolver
+            ->setRequired('id')->setAllowedTypes('id', ['string'])
+            ->setRequired('status')->setAllowedValues('status', [self::STATUS_PENDING, self::STATUS_INVALID, self::STATUS_VALID])
+            ->setDefault('reason', null)->setAllowedTypes('reason', ['int', 'null'])
+            ->setDefault('reasonMessage', null)->setAllowedTypes('reasonMessage', ['string', 'null'])
+            ->setDefault('result', false)->setAllowedTypes('result', ['array', 'null'])
```
Comment (konandrum) [inline Model/ValidatorResult.php]: default False, but allowedTypes missing boolean ?

**Review by konandrum — CHANGES_REQUESTED** [review]

File: Model/ValidatorResult.php (hunk 3 above)
Comment (Gwemox) [reply Model/ValidatorResult.php]: Changed to null

File: Model/MatchingResult.php (hunk 1 above)
Comment (Gwemox) [reply Model/MatchingResult.php]: Fix

File: Model/MatchingResult.php (hunk 2 above)
Comment (Gwemox) [reply Model/MatchingResult.php]: Fix

**Pull request merged**